File: ordenacoes/insertionsort.py
# Estrutura de Dados II
# Matheus Marinho
# Insertion Sort

def insertionSort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and key < arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key
    return arr

# Input:    
arr = [50, 16, 23, 5, 7, 12, 10]

# Output:
# Array original:  [50, 16, 23, 5, 7, 12, 10]
# Array ordenado:  [5, 7, 10, 12, 16, 23, 50]

print("Array original: ", arr)
print("Array ordenado: ", insertionSort(arr))
print("Menor valor: ", arr[0])
print("Maior valor: ", arr[len(arr)-1])

File: ordenacoes/insertionsort.c
/* Estrutura de Dados II
Matheus Marinho
Insertion Sort */

#include <stdio.h>
#include <stdlib.h>

void insertionSort(int arr[], int tam){
    int chave, j;
    for(int i = 1; i < tam; i++){
        chave = arr[i];
        j = i - 1;
        while (j >= 0 && arr[j] > chave){
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = chave;        
    }
}

int main(){
    int arr[] = {50, 16, 23, 5, 7, 12, 10};
    int size = sizeof(arr)/sizeof(arr[0]);
    
    printf("Vetor original: ");
    for(int i = 0; i < size; i++){
        printf("%d ", arr[i]);
    }
    printf("\n");
    
    // Chamando a função para ordenar o array
   insertionSort(arr, size); 
    
    printf("Vetor ordenado: ");
    for(int i = 0; i < size; i++){
        printf("%d ", arr[i]);
    }
    printf("\n");

    printf("Menor valor do array: [%d]", arr[0]);
    printf("\n");
    printf("Maior valor do array: [%d]", arr[size-1]);
    return 0;
}

